flutter_reactive_ble_platform_interface/lib/src/reactive_ble_platform_interface.dart

```diff
@@ -0,0 +1,120 @@
+import 'package:flutter_reactive_ble_platform_interface/src/plugin_controller.dart';
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+
+import 'models.dart';
+
+abstract class ReactiveBlePlatform extends PlatformInterface {
+  ReactiveBlePlatform() : super(token: _token);
+  static final Object _token = Object();
+
+  static ReactiveBlePlatform _instance =
+      const PluginControllerFactory().create();
+
+  static ReactiveBlePlatform get instance => _instance;
+
+  /// Platform-specific plugins should set this with their own platform-specific
+  /// class that extends [ReactiveBlePlatform] when they register themselves.
+  static set instance(ReactiveBlePlatform instance) {
+    PlatformInterface.verifyToken(instance, _token);
+    _instance = instance;
+  }
+
+  Stream<ScanResult> get scanStream {
+    throw UnimplementedError('scanStream has not been implemented.');
+  }
```

Comment:
All this stuff looks weird to me. What's the point of providing a throwing implementation? Why trying to stuff a singleton in what should presumably be an interface? What's that shady business with the token? Why to for extending this class instead of implementing an interface?

In short, if this structure is not enforced by Flutter, I suggest to clean it up.

Reply:
The token is enforced by flutter and the not implemented error is also a common practice promoted by google: "The last thing to note is that all of the methods in the platform interface should have a default implementation that just throws an UnimplementedError. Since every implementation of UrlLauncherPlatform must use extends, then if a new method is added to the interface, it won’t be a breaking change that causes apps to break"